lib/Graph/Graph.cpp

```diff
@@ -1539,15 +1545,28 @@ Function::createFusedRowwiseQuantizedSparseLengthsWeightedSum(
       this, name, data, weights, indices, lengths);
 }
 
-FusedRowwiseQuantizedSparseLengthsWeightedSumNode *
+FusedRowwiseQuantizedSparseLengthsSumNode *
 Function::createFusedRowwiseQuantizedSparseLengthsSum(llvm::StringRef name,
                                                       Tensor &data,
                                                       NodeValue indices,
                                                       NodeValue lengths) {
-  auto ty = getParent()->uniqueType(ElemKind::FloatTy, {indices.dims()[0]});
-  auto ones = createSplat(name.str() + ".ones", ty, 1.0);
-  return quantizeDataAndCreateFusedRowwiseQuantizedSparseLengthsWeightedSum(
-      this, name, data, ones, indices, lengths);
+  // For fused rowwise quantization, we must have a two-dimensional input. If
```

Comment:
Same as other -- can we try to share as much of this logic as possible across RWQSLS/RWQSLWS, e.g. in static helpers here in Graph.cpp?

Reply:
Done with modifying quantizeData...(...) a little but. Let me know you think it's OK.